.github/workflows/get_ica_access_token.yml

```diff
@@ -0,0 +1,105 @@
+name: get-encrypted-ica-access-token
+
+on:
+  workflow_call:
+    inputs:
+      role_to_assume_arn:
+        required: true
+        type: string
+      secret_arn:
+        required: true
+        type: string
+      project_id:
+        required: true
+        type: string
+      aws_region:
+        required: true
+        type: string
+      branch_name:
+        required: true
+        type: string
+    secrets:
+      symmetrical_encryption_key:
+        required: true
+
+    outputs:
+      ica_access_token_encrypted:
+        description: the ica project access token (encrypted with the symmetrical encryption key)
+        value: ${{ jobs.get_token.outputs.ica_access_token_encrypted }}
+
+permissions:
+  id-token: write   # This is required for requesting the JWT
+  contents: read    # This is required for actions/checkout
+
+jobs:
+  get_token:
+    name: get-token
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      # Configure creds
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ inputs.role_to_assume_arn }}
+          aws-region: ${{ inputs.AWS_REGION }}
+
+      # Retrieve JSON secret (must parse out as json)
+      - name: Retrieve Secret
+        uses: aws-actions/aws-secretsmanager-get-secrets@v1
+        with:
+          secret-ids: |
+            ${{ inputs.secret_arn }}
+          parse-json-secrets: true
+
+      # Get env var for secret
+      # SECRET will be set as env var 'ICASECRETSWORKFLOW_PROJECT_ID'
```

Comment:
Can you explain this a bit? Does the `Retrieve Secret` step export the secret in json format under the name `ICASECRETSWORKFLOW_PROJECT_ID` and then you just assign that to the `secret_env_var` variable? Or something else going on here..

Reply:
This was tricky, the AWS secret value is a json string and is of the form

```
{
  "my_project_id": "ejyra..."
}
```

This GitHub action however santises the secerts and removes any double quotes, so it's no longer a json string.  

By adding in the parse_json parameter, it instead creates env variables like ICASECRETSWORKFLOW_DDDD6C29_24D3_49F4_91C0_7E818B3C0A21